### PR TITLE
core: Remove morph shape preloading

### PR DESCRIPTION
**Pull request by Herschel**
Remove the preload step that would pre-create the shapes for each morph shape ratio on SWF load. Instead, lazily crate the shapes as they are needed. We already do similar for the drawing API, so it doesn't seem necessary to create these shapes ahead of time.

This removes a large amount of complex code from `MovieClip::preload`, as all timelines had to be scanned to determine what ratios of a morph shape were used.

### CODE REVIEW

**Review by adrian17 — APPROVED** [review]
Patch itself LGTM.
Questions:
- I'm assuming this partially unblocks the async preload PR? (it had issues with morph shapes)
- Did you test perf on some morphshape-heavy movie? With our tessellation being relatively slow, better make sure it's not slow enough to make things unplayable

File: core/src/display_object/morph_shape.rs [hunk 1]
```diff
@@ -39,8 +44,9 @@ impl<'gc> MorphShape<'gc> {
         self.0.read().ratio
     }
 
-    pub fn set_ratio(&mut self, gc_context: MutationContext<'gc, '_>, ratio: u16) {
-        self.0.write(gc_context).ratio = ratio;
+    pub fn set_ratio(&mut self, context: &mut UpdateContext<'_, 'gc, '_>, ratio: u16) {
```
Comment (relrelb) [inline core/src/display_object/morph_shape.rs:47]: What's the motivation for this change? `set_ratio` doesn't really need `context`, but just `gc_context`.

Comment (Herschel) [reply core/src/display_object/morph_shape.rs:47]: Good catch, this was accidental (originally I had lazily initialized whenever the ratio changed, but switched to doing it on render/getting bounds).

**Comment by Herschel** [comment]
> Patch itself LGTM. Questions:
> 
> * I'm assuming this partially unblocks the async preload PR? (it had issues with morph shapes)

Yep, that's one motivation. In general I want to get rid of the hacky junk in `preload` that has always caused headaches -- similarly I'd like to remove the `preload_sound_stream` fns.

> * Did you test perf on some morphshape-heavy movie? With our tessellation being relatively slow, better make sure it's not slow enough to make things unplayable

Tested a lot of NG movies last night and seems good in practice. I can definitely make a giant morph shape that will tank FPS as it first tessellates, but I think this is preferable to spending the same amount of time time tessellating all frames on load. I'll take some measurements.

This is similar to what we do with the drawing API for games like Interactive Buddy. I think this is the right path forward in general -- to allow for proper rendering of strokes and avoid jaggies on zoom, we'll eventually want to tessellate most things on demand regardless with proper caching.

File: core/src/display_object/morph_shape.rs [hunk 2]
```diff
@@ -21,15 +22,17 @@ pub struct MorphShapeData<'gc> {
 }
 
 impl<'gc> MorphShape<'gc> {
-    pub fn new(
-        gc_context: gc_arena::MutationContext<'gc, '_>,
-        static_data: MorphShapeStatic,
+    pub fn from_swf_tag(
+        context: &mut UpdateContext<'_, 'gc, '_>,
```
Comment (relrelb) [inline core/src/display_object/morph_shape.rs:26]: I guess `context` can be downgraded back to `gc_context`?
